content/cucumber/hooks.md

```diff
@@ -76,71 +148,136 @@ end
 end
-```
-
-Pass `AND` Tags as separate strings:
 
-```ruby
+# Pass `AND` Tags as separate strings:
 Before('@cucumis', '~@sativus') do
   # This will only run before scenarios tagged
   # with @cucumis AND NOT @sativus.
-end
-```
 
-You create complex Tag conditions using both `OR` and `AND` on Tags:
-
-```ruby
+# You create complex Tag conditions using both `OR` and `AND` on Tags:
 Before('@cucumis, @sativus', '@aqua') do
   # This will only run before scenarios tagged
   # with (@cucumis OR @sativus) AND @aqua
 end
-```
-
-`AfterStep` example:
 
-```ruby
+# `AfterStep` example:
 AfterStep('@cucumis', '@sativus') do
   # This will only run after steps within scenarios tagged
   # with @cucumis AND @sativus.
 end
+
+# Pass `NOT` Tag
+Before('NOT @cucumis') do
+  # This will only run before scenarios NOT tagged
+  # with @cucumis
+end
+```
+
+```java
+@Before
+public void beforeScenario(){
+    // This hook will be executed before all scenarios
+}
+
+@Before("@foo")
+public void beforeScenario(){
+    // This hook will be executed before scenarios tagged with @foo
+}
+
+@Before("@foo and @bar")
+public void beforeScenario(){
+    // This hook will be executed before scenarios tagged with @foo and @bar
+}
+
+@Before("@foo or @bar")
+public void beforeScenario(){
+    // This hook will be executed before scenarios tagged with @foo or @bar
+}
+
+@Before("not @foo")
+public void beforeScenario(){
+    // This hook will be executed before scenarios NOT tagged with @foo
+}
+```
+
+```javascript
+var {defineSupportCode} = require('cucumber');
+
+defineSupportCode(function({After, Before}) {
+  Before(function () {
+    // This hook will be executed before all scenarios
+  });
+
+  Before({tags: "@foo"}, function () {
+    // This hook will be executed before scenarios tagged with @foo
+  });
+
+  Before({tags: "@foo and @bar"}, function () {
+    // This hook will be executed before scenarios tagged with @foo and @bar
+  });
+
+  Before({tags: "@foo or @bar"}, function () {
+    // This hook will be executed before scenarios tagged with @foo or @bar
+  });
+
+  Before({tags: "not @foo"}, function () {
+    // This hook will be executed before scenarios NOT tagged with @foo
+  });
+
+  // You can use the following shorthand when only specifying tags
+  Before("@foo", function () {
+    // This hook will be executed before scenarios tagged with @foo
+  });
+});
 ```
 
+See more documentation on [tag expressions](https://docs.cucumber.io/tag-expressions/).
+
 **Think twice before you use Hooks!** 
 Whatever happens in Hooks is invisible to people who only read the Features. 
-You should consider using [Background](/gherkin/background/) as a more explicit 
-alternative, especially if the setup should be readable by non-technical people.
+Consider using [Background](/gherkin/background/) as a more explicit alternative,
+especially when the setup should be readable by non-technical people.
 
 ## Global Hooks
 
+{{% text "ruby" %}}
```

Comment:
A note on the paragraph after this `{{% text "ruby" %}}` annotation: Can we somehow get rid of the "just" in that sentence?